src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -164,50 +163,32 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(DbConnectionValidat
         return DbConnectionValidatorResult.DocumentTableFound;
     }
 
-    private static ISqlBuilder GetSelectBuilderForDocumentTable(ISqlBuilder sqlBuilder, string documentTable, string schema)
+    private static string GetDocumentCommandText(ISqlBuilder sqlBuilder, string documentTable, string schema, bool isShellDescriptorDocument = false)
     {
         sqlBuilder.Select();
         sqlBuilder.Selector("*");
         sqlBuilder.Table(documentTable, alias: null, schema);
-        sqlBuilder.Take("1");
 
-        return sqlBuilder;
-    }
+        if (isShellDescriptorDocument)
+        {
+            sqlBuilder.WhereAnd($"Type = '{_shellDescriptorTypeColumnValue}'");
+        }
 
-    private static ISqlBuilder GetSelectBuilderForShellDescriptorDocument(ISqlBuilder sqlBuilder, string documentTable, string schema)
-    {
-        sqlBuilder.Select();
-        sqlBuilder.Selector("*");
-        sqlBuilder.Table(documentTable, alias: null, schema);
-        sqlBuilder.WhereAnd($"Type = '{_shellDescriptorTypeColumnValue}'");
         sqlBuilder.Take("1");
 
-        return sqlBuilder;
+        return sqlBuilder.ToString();
     }
 
-    private static IConnectionFactory GetFactory(string databaseProvider, string connectionString)
-    {
-        return databaseProvider switch
-        {
-            DatabaseProviderValue.SqlConnection => new DbConnectionFactory<SqlConnection>(connectionString),
-            DatabaseProviderValue.MySql => new DbConnectionFactory<MySqlConnection>(connectionString),
-            DatabaseProviderValue.Sqlite => new DbConnectionFactory<SqliteConnection>(connectionString),
-            DatabaseProviderValue.Postgres => new DbConnectionFactory<NpgsqlConnection>(connectionString),
-            _ => throw new ArgumentOutOfRangeException(nameof(databaseProvider), "Unsupported database provider"),
-        };
-    }
-
-    private static ISqlDialect GetSqlDialect(string databaseProvider)
-    {
-        return databaseProvider switch
+    private static (IConnectionFactory connectionFactory, ISqlDialect sqlDialect) GetFactoryAndSqlDialect(
+        string databaseProvider,
+        string connectionString) => databaseProvider switch
```

Comment:
I'd agree since I prefer expression-bodied members, but in OC we deliberately don't use them (and it's a bit easier to read in this case too).